pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -1324,15 +1394,21 @@ class Build {
                         throw new Exception("[ERROR] Sign job timeout (${buildTimeouts.SIGN_JOB_TIMEOUT} HOURS) has been reached OR the downstream sign job failed. Exiting...")
                     }
                 }
-
-                if (enableTests && buildConfig.TEST_LIST.size() > 0) {
-                    try {
-                        // Run tests if we have a test list, don't use timeouts as the jobs have their own
-                        def testStages = runTests()
-                        context.parallel testStages
-                    } catch (Exception e) {
-                        context.println "Failed test: ${e}"
+                
+                // Run Smoke Tests and AQA Tests
+                try { 
+                    runSmokeTests()
```

Comment:
Do we always want to have smoke tests enabled? For those who may not want to run said tests because they only want to run a quick test build or have no need to run these tests (i.e. third parties who wish to build using our scripts) then this is not ideal. I would prefer to have it inside the `enableTests && buildConfig.TEST_LIST.size() > 0` conditional.

Reply:
If those tests fail, the build is in shambles. `openjdk-build` didn't do what it was supposed to do. That's the reason those tests are in `openjdk-build` and not over in `openjdk-tests`. The number of tests is small and a test run is quick (less than a minute). So there should not be a reason not to run these tests and it's the reason why they are a hard failure contrary to our normal tests that are soft.

There could be an escape hatch, but it may not be `buildConfig.TEST_LIST.size()` (or anything else used to normally skip the tests) if it comes from this line:

https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/blob/b50b6ab0344d2c55c19863aca3c6e22e23f2db22/pipelines/jobs/configurations/jdk17_pipeline_config.groovy#L7

Because we have specialists that disable those tests, although, in that case, the build should not have been enabled in the first place.

Reply:
Resolved by https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/pull/86#issuecomment-812107292